src/amp-story-player/amp-story-player-impl.js

```diff
@@ -705,20 +734,43 @@ export class AmpStoryPlayer {
    * @param {number} storyDelta
    */
   go(storyDelta) {
-    if (
-      this.currentIdx_ + storyDelta >= this.stories_.length ||
-      this.currentIdx_ + storyDelta < 0
-    ) {
-      throw new Error('Out of Story range.');
-    }
     if (storyDelta === 0) {
       return;
     }
-
-    const currentStory = this.stories_[this.currentIdx_ + storyDelta];
-
-    this.show(currentStory.href);
-    this.signalNavigation_();
+    if (this.element_.getAttribute('enable-circular-wrapping') !== 'true') {
```

Comment:
To avoid nested `if` conditionals, you can extract them to their parent `if` and do an early return. 
```suggestion
    if (this.element_.getAttribute('enable-circular-wrapping') !== 'true' && this.isIndexOutOfBounds_(index)) {
      throw new Error('New story index is out of bounds.');
    }
```

I think repeating this for the other nested conditions will help us do some clean up. It also helps if we can extract repeated logic into a method that can be shared.

Reply:
would`this.show(currentStory.href);
this.signalNavigation_();` be repeated logic?

Reply:
Yeah, I think we can encapsulate that to another method